movies.py

```diff
@@ -15,3 +15,10 @@ class Movie(BaseModel):
 
 async def get_all_movies() -> list[Movie]:
     return movies
+
+
+async def get_movie_by_name(name: str) -> Movie | None:
```

Comment:
For a better separation of concerns (and easier testability), I'd change this function signature to accept the list of movies as an argument, instead of calling `get_all_movies`.
This would also have the added benefit of making it not async, which is not a benefit per-se, but is a good indicator.

Reply:
Won't this move the responsibility to the web server to pass the argument? Is this best practice?

Reply:
I usually try to keep functions with a single responsiblity.
Currently this function is responsible for 1. fetching movies and 2. searching.
This also limits the search usability, imagine you have some other way of getting movies (from cache, as input, as a result some other processing, etc.), you wouldn't be able to reuse the search logic, since it is currently coupled to a specific fetching method.

Regarding the server - I'd do it like that, yes, I think it's fine.
If that bothers you I guess you can wrap it in some `fetch_and_filter` function that both fetches and filters, but I wouldn't bother.

Reply:
I understand.

This also happens because I'm not using a DB so currently to way to query is to get all movies and then filter